### ISSUE TITLE
installation issues with pip

### ISSUE THREAD

**Issue by mrocklin** [uqfoundation/pathos#2]
```
(test-pathos)mrocklin@linux2:~$ pip install pathos
Downloading/unpacking pathos
  Could not find a version that satisfies the requirement pathos (from versions: 0.1a1)
Cleaning up...
No distributions matching the version for pathos
Storing complete log in /home/mrocklin/.pip/pip.log
```

**Comment by jazzblue** [comment]
Hi, 
I am doing  **pip install git+https://github.com/uqfoundation/pathos**, but am getting the following error (see below in the end in **bold**).

How can I get around it.

Downloading/unpacking git+https://github.com/uqfoundation/pathos
  Cloning https://github.com/uqfoundation/pathos to c:\users\gronin\appdata\loca
l\temp\pip-jldriz-build
  Running setup.py (path:c:\users\gronin\appdata\local\temp\pip-jldriz-build\set
up.py) egg_info for package from git+https://github.com/uqfoundation/pathos

Downloading/unpacking pp>=1.6.4.4 (from pathos==0.2a1.dev)
  **Could not find any downloads that satisfy the requirement pp>=1.6.4.4 (from pathos==0.2a1.dev)**

**Comment by mmckerns** [comment]
I'm pushing to get a new `pip`-friendly release out this month.  You should be able to workaround by applying the "prerelease" flag (i.e. `--pre`) for `pip`.

**Comment by jazzblue** [comment]
Seems like I am having problem with installing processing-0.52-pathos on my windows 64. I am getting:

**ValueError: [u'path']**

Are you planning to have win54 support? Thanks.

**Comment by mmckerns** [comment]
Windows 64 is supported wherever possible, however I don't develop or test regularly on Windows, so I may not catch everything that's a windows issue until I test on Windows around a release. Can you open this as a new ticket, where you paste the traceback you are getting?

**Comment by jazzblue** [comment]
opened: https://github.com/uqfoundation/pathos/issues/49
Thanks.

**Comment by mmckerns** [comment]
See #52, #53, and #54 for WIP to solve the `pip` install issue.

**Comment by mmckerns** [comment]
Apparently something like this currently works… `sudo pip install git+https://github.com/uqfoundation/mystic.git@master`… if you install any uqfoundation dependencies first.  Progress still needs to be made on #52, #53, #54 for automated dependency install.

**Comment by mmckerns** [comment]
This works: 
`pip-2.7 install --target=test git+https://github.com/uqfoundation/pathos.git@master`,
without  downloading and installing any dependencies first.

Note that if you have pre-existing `pp` installations, a new `ppft` may require manually deleting some files (as the `pp` install is nonstandard and `ppft` "corrects" that), and also if you have `processing` installed, then `pathos` will skip the `multiprocess` install.  So, a little work for current users, but for newbies, it's a one-line `pip` install.

No blockers to a new release (#43) now.

**Comment by mmckerns** [comment]
Ah… it does *install* just fine.  However, due to how `multiprocess` was built, there may be limited functionality if `lib` is before `lib/site-packages` in your PYTHONPATH. See [multiprocess ticket #3](https://github.com/uqfoundation/multiprocess/issues/3).

**Comment by mmckerns** [comment]
I just updated everything in the critical path, and pathos now installs cleanly and also works on windows.  Only thing left is  change the version number and release.  It should install with pip from master w/o manually downloading any dependencies, and on all platforms.

**Comment by chrisidefix** [comment]
It looks good, but I would avoid installing with `sudo`. Instead I would recommend this, which worked for me: `pip install --user git+https://github.com/uqfoundation/pathos`

**Comment by The3DWizard** [comment]
Just installed it via `pip install --user git+https://github.com/uqfoundation/pathos` without any problems. Thank you very much.

**Comment by mmckerns** [comment]
This also should work:  `pip install git+https://github.com/uqfoundation/pathos.git@master`… any variants like `--user` or `--install-option='--prefix'` are just where it's going to be installed, and they should all work as expected.  As soon as the release is out, then `pip install pathos`, and all of the variants of that will then work.

**Comment by chrisidefix** [comment]
Thanks, I merely wanted to point out that it is generally not a good idea to install python packages with `sudo` privileges. It often creates trouble when you want to execute the code, but of course it all depends on your individual configuration. Looking forward to seeing a `pypi` release!

**Comment by miguelfg** [comment]
I've installed with: pip install git+https://github.com/uqfoundation/pathos.git@master  and worked ok, and it has the multiprocessing module in, while via pip or easy_install didn't.
But i think as i'm using virtualenv, now it's loading the standard multiprocessing  as it's said in https://github.com/uqfoundation/multiprocess/issues/3

Any help for this?

**Comment by mmckerns** [comment]
@miguelfg: Can you give more detail on what you are seeing?  I don't understand what you are seeing from your comment.  Also feel free to open up a new ticket for your issue.

**Comment by miguelfg** [comment]
I've confused issues. I've pointed to an issue of another of your libraries (multiprocessing), because it might happened that to me. After successfully installed pathos as described, I've re run my script and still raised my previous problem:

> python PicklingError: Can't pickle <type 'thread.lock'>: attribute lookup thread.lock failed

That's why I've installed pathos, to run multiprocessing. Since pickling kept failing with pathos too, I thought it could be related to the issue of your multiprocessing library/repo, when '/lib' is in the path of my virtualenv site-packages folder it keeps loading the standard multiprocessing and not pathos's one (as said in the issue).
However I've ended refactoring code, and moving my multiprocessing method outside the class where it was, and not dependent to staticmethods that it was calling. After this, the standard multiprocessing and also pathos work fine.
